tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php

```diff
@@ -828,20 +841,23 @@ private function setUpSearchRequestMocks(
             ['flSmartDidYouMean', $smartDidYouMean],
         ];
 
-        $contextMock = $this->getMockBuilder(Context::class)->disableOriginalConstructor()->getMock();
-
-        if ($withSmartDidYouMean) {
-            $contextMock->expects($this->any())->method('addExtension')->withConsecutive(
-                ['findologicService', $findologicService],
-                ['flSmartDidYouMean', $smartDidYouMean]
-            );
-        } else {
-            $contextMock->expects($this->any())->method('addExtension')->withConsecutive(
-                ['findologicService', $findologicService]
-            );
+        if (!$context) {
+            $context = $this->getMockBuilder(Context::class)->disableOriginalConstructor()->getMock();
+            $context->expects($this->any())->method('getExtension')->willReturnMap($defaultExtensionMap);
+
+            if ($withSmartDidYouMean) {
+                $context->expects($this->any())->method('addExtension')->withConsecutive(
+                    ['findologicService', $findologicService],
+                    ['flSmartDidYouMean', $smartDidYouMean]
+                );
+            } else {
+                $context->expects($this->any())->method('addExtension')->withConsecutive(
```

Comment:
Should be just `->with()` instead of `->withConsecutive()`

Reply:
True, in this case it doesn't make sense to use `withConsecutive`.

Reply:
I tried using `with`, but it does not seem to work, since other tests use `withConsecutive` on the same method and using `withConsecutive` multiple times on the same method, causes them to merge together.

E.g.

```php
$someMock->expects($this->any())->method('foo')->withConsecutive('bar');
$someMock->expects($this->any())->method('foo')->withConsecutive('foobar');
```

Will actually make the first access return `foo` and the second access `foobar`. Using only `with`, won't allow this merging and it therefore does not work.

Reply:
That might lead to confusion but we leave it for now, same logic was already there before.